File: project/golang/unbounded-cpu.go
package main

// import the required packages
import "fmt"
import "os"
import "time"

func main() {
	// declare a done variable that is a boolean
	var done bool
	// indicate that the program is not done running
	done = false
	// extract the command-line argument(s)
	var prog_name = os.Args[0]
	var entered_value = os.Args[1]
	// always iterate in an unbounded-fashion
	for !done{
		// TODO: make sure that the program output matches the README file
		time.Sleep(1*time.Second)
		fmt.Println("Run", prog_name, "with", entered_value)
	}
}

File: project/clang/unbounded-cpu.c
// Add required imports
#include <stdio.h>

int main(int argc, char *argv[])
{
  // confirm that there are two command-line arguments
  if (argc == 2)
  {
    // extract the name of the program from command-line arguments
    // read the second command line argument which is a letter
    // always print a value in an non-bounded iteration construct
    while (1)
    {
      // make sure that the program output matches the README file
      printf("Run %s with %s \n", argv[0], argv[1]);
    }
  }
  else
  {
    printf("Please enter two command-line arguments, one being the file name.");
  }
}
